### PR TITLE
Mm 20561: Updating admin manage roles modal to typescript

### PR DESCRIPTION
**Pull request by opllama2**
#### Summary
This PR updates the admin console manage roles modal to use typescript within `components/admin_console/manage_roles_modal`

#### Ticket Link
Jira Link: https://mattermost.atlassian.net/browse/MM-20561
Fixes https://github.com/mattermost/mattermost-server/issues/13511

### CODE REVIEW

File: components/admin_console/manage_roles_modal/manage_roles_modal.tsx [hunk 1]
```diff
@@ -14,8 +13,27 @@ import {trackEvent} from 'actions/diagnostics_actions.jsx';
 import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
 import BotBadge from 'components/widgets/badges/bot_badge';
 import Avatar from 'components/widgets/users/avatar';
+type Props = {
+    show: boolean;
+    user?: any;
```
Comment (wiggin77) [inline components/admin_console/manage_roles_modal/manage_roles_modal.tsx]: Quick question:  is the intention to make `user` a specific type when more code gets ported to TS?

Comment (opllama2) [reply components/admin_console/manage_roles_modal/manage_roles_modal.tsx]: thats a good question but i dont know about a plan to do so, imo its a good idea to do so if there's any "starter" type for the `user` object i can change to that type and we can append to it as we go.

**Comment by hanzei** [comment]
@opllama2 Sorry for the delay with the reviews. Someone will look into your PR over the next few days.

**Comment by opllama2** [comment]
Thnx all, proud of it my first contribution 🥰

**Pull request merged**